### PR TITLE
Update efs-provisioner to the latest version

### PR DESCRIPTION
**Pull request by linki**
Includes aws sdk with support for webidentity. Also some rbac bix :astonished:

### CODE REVIEW

File: cluster/manifests/deletions.yaml [hunk 1]
```diff
@@ -28,3 +28,6 @@ post_apply:{{ if and (ne .ConfigItems.teapot_admission_controller_process_resour
 - name: ingress-template-controller
   kind: ClusterRoleBinding
 {{ end }}
+- name: leader-locking-efs-provisioner
+  namespace: default
+  kind: Role
```
Comment (mikkeloscar) [inline cluster/manifests/deletions.yaml]: kind should be `RoleBinding`

Comment (linki) [reply cluster/manifests/deletions.yaml]: Thanks!

**Comment by linki** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**